### PR TITLE
Inline fix logic

### PR DESCRIPTION
**Pull request by wraithgar**
- deps: add semver@7.5.3
- deps: add hosted-git-info@6.1.1
- fix: pull in fix logic from normalize-package-data
- fix: only report bundleDependencies change if it was changed

### CODE REVIEW

File: lib/normalize.js [hunk 1]
```diff
@@ -6,6 +7,27 @@ const legacyMakeWarning = require('normalize-package-data/lib/make_warning.js')
...
+  return !spec.match(/[/@\s+%:]/) &&
+    spec === encodeURIComponent(spec)
+}
+
+function isValidScopedPackageName (spec) {
+  if (spec.charAt(0) !== '@') {
+    return false
+  }
+
+  var rest = spec.slice(1).split('/')
```
Comment (lukekarrys) [inline lib/normalize.js:22]:
```suggestion
  const rest = spec.slice(1).split('/')
```

File: lib/normalize.js [hunk 2]
```diff
@@ -70,7 +125,7 @@ const normalize = async (pkg, { strict, steps, root, changes, allowLegacyCase })
         changes?.push(`"bundleDependencies" was changed from an object to an array`)
         data.bundleDependencies = Object.keys(bd)
       }
-    } else {
+    } else if ('bundleDependencies' in data) {
```
Comment (lukekarrys) [inline lib/normalize.js:128]:
```suggestion
    } else if (Object.prototype.hasOwnProperty.call(data, 'bundleDependencies')) {
```
to be consistent with other data checks in fixer

**Comment by lukekarrys** [comment]
Looks good! One minor swap for a `var` -> `const` and a suggestion.

File: lib/normalize.js (hunk 2 above)
Comment (wraithgar) [reply lib/normalize.js:128]: https://github.com/npm/package-json/blob/main/lib/normalize.js#L304

We do it both ways an `in` is way easier to read imho.

**Review by lukekarrys — APPROVED** [review]

File: lib/normalize.js [hunk 3]
```diff
@@ -18,17 +40,49 @@ const normalize = async (pkg, { strict, steps, root, changes, allowLegacyCase })
...
 
   if (steps.includes('fixVersionField') || steps.includes('normalizeData')) {
-    legacyFixer.fixVersionField(data, strict)
+    // allow "loose" semver 1.0 versions in non-strict mode
+    // enforce strict semver 2.0 compliance in strict mode
+    const loose = !strict
+    if (!data.version) {
+      data.version = ''
+    } else {
+      if (!semver.valid(data.version, !loose)) {
```
Comment (hashtagchris) [inline lib/normalize.js:77]: (Here and line 80) bool arg is `loose` based on typings: https://unpkg.com/browse/@types/semver@7.3.10/functions/valid.d.ts

```suggestion
      if (!semver.valid(data.version, loose)) {
```

Comment (wraithgar) [reply lib/normalize.js:77]: Yep this is a typo from refactoring, I'd changed it to !strict but then decided to keep the loose param.

**Pull request merged**